src/connect.ts

```diff
@@ -1,10 +1,59 @@
+import {
+  Manifest,
+  FlashState,
+  FlashError,
+  FlashStateType,
+} from "./const.js";
 import type { InstallButton } from "./install-button.js";
 import "./install-dialog.js";
 
-export const connect = async (button: InstallButton) => {
+export const connect = async (
+  onEvent: (state: FlashState) => void,
+  button: InstallButton,
+  ) => {
   let port: SerialPort | undefined;
+  let manifest: Manifest;
+
+  const fireStateEvent = (stateUpdate: FlashState) =>
+  onEvent({
+    ...stateUpdate,
+    manifest
+  });
+
+  let serialPortFilter: SerialPortFilter[] | undefined
+  let serialPortRequestOptions: SerialPortRequestOptions
+
+  const el = document.createElement("ewt-install-dialog");
+  el.manifestPath = button.manifest || button.getAttribute("manifest")!;
```

Comment:
We shouldn't pass the manifest path to the element anymore. Just pass the manifest.

Reply:
There are no duplicate variables here, I just moved up the manifest path of the element before the request port routine to aid in fetching.

The manifestPath of the element is used throughout other parts of the project, for flashing. Changing this would mean changing everywhere else unnecessarily. Please check again.